### PR TITLE
Use Phosphor Datagrid for CSV Viewer

### PR DESCRIPTION
**Pull request by blink1073**
Removes the restriction on the number of renderable rows (from 1000 to the integer overflow value).

<image src="https://user-images.githubusercontent.com/2096628/27061731-e7792d1a-4fab-11e7-9ae2-f225b40c82de.png" width=400>

### CODE REVIEW

**Comment by sccolbert** [comment]
For the record, integer "overflow" occurs at `9007199254740991`.

I dunno, that may not be enough...

**Comment by sccolbert** [comment]
You may consider setting `.headerVisibility = 'column'` to hide the row headers.

**Comment by sccolbert** [comment]
Looks like the data format might be amenable to using the `JSONModel` instead of rolling your own.

**Comment by blink1073** [comment]
Playing with it, I actually like seeing which row I'm on.

**Comment by blink1073** [comment]
I looked into using `JSONModel`: we'd have to install a new one whenever the delimiter changes.

**Comment by blink1073** [comment]
Removed the row number indicator to conform with what we had previously.

**Comment by sccolbert** [comment]
I wonder if we should have a `get/set data` property on `JSONModel` then. Although, what's the use case for changing the delimiter in a CSV file. Shouldn't that be known up front?

**Comment by blink1073** [comment]
Not necessarily, files can be called csv and use other delimiters.

**Comment by sccolbert** [comment]
I meant for a *given* file, is there ever a need to change the delimiter?

**Comment by sccolbert** [comment]
Also, swapping data models on the grid is no more expensive than emitting `model-reset`.

**Comment by blink1073** [comment]
Switched to the `JSONModel`.  The use case for the set-able delimiter in the toolbar is opening a file that you think is CSV only to release it is tab or semi-colon separated.

File: packages/csvviewer/src/widget.ts [hunk 1]
```diff
@@ -99,36 +97,23 @@ class CSVViewer extends Widget {
...
-    this._model = null;
-    this._table = null;
+    this._grid = null;
     this._toolbar = null;
-    this._warning = null;
     this._monitor = null;
 
-    model.dispose();
-    table.dispose();
+    grid.dispose();
```
Comment (sccolbert) [inline packages/csvviewer/src/widget.ts]: You don't need to dispose of your child widgets. The layout does it automatically.

**Pull request merged**